OpenRA.Game/Network/Order.cs

```diff
@@ -147,16 +152,26 @@ public static Order Deserialize(World world, BinaryReader r)
 
 						var targetString = flags.HasField(OrderFields.TargetString) ? r.ReadString() : null;
 						var queued = flags.HasField(OrderFields.Queued);
+
+						List<Actor> extraActors = null;
+						if (flags.HasField(OrderFields.ExtraActors))
+						{
+							var count = r.ReadInt32();
```

Comment:
Maybe missing something but could this not be simplified to 

```suggestion
							var count = r.ReadInt32();
							extraActors = new List<Actor>(count);
							for (int i = 0; i < count; i++)
							{
								var actorId = r.ReadUInt32();
								extraActors.Add(world.GetActorById(actorId));
							}
```

Reply:
And add the world null check that i removed!

Reply:
How is that simpler? I'd rather leave the `if` statement outside of the `for` loop.

Reply:
Imo because we are doing it in one sweep instead of two (first creating a list of Ids then retrieve the actors with the Ids)

Reply:
Fixed.